src/partition/growfs.c

```diff
@@ -92,8 +92,7 @@ static int maybe_resize_underlying_device(const char *mountpath, dev_t main_devn
         cryptsetup_enable_logging(NULL);
 #endif
 
-        r = get_block_device_harder(mountpath, &devno);
-        if (r < 0)
+        if(get_block_device_harder(mountpath, &devno) <= 0)
                 return log_error_errno(r, "Failed to determine underlying block device of \"%s\": %m",
```

Comment:
So there *is* a problem here: when `get_block_device_harder` returns 0, devno==0, and we should not proceed to treat this is a useful device number. We shouldn't treat that as an error either. So something like this:
```C
r = get_block_device_harder(mountpath, &devno);
if (r < 0)
        return log_error_errno(r, "Failed to determine underlying block device of \"%s\": %m",
                                            mountpath);
if (r == 0)
         return log_debug("Mountpath %s has no underlying block device, ignoring.", mountpath);
```

It would be also good to look at the other places where get_block_device_harder or get_block_device are used and verify that a similar issue does not occur.

(What the commit message says about uninitialiazed `devno` is not true.)

Reply:
Thanks for the review comments. I am working on it.

Reply:
working on review comments in progress.

Reply:
Hi @keszybz , fixed review comments. Kindly have a look.